### ISSUE TITLE
Change the way of adopting Theia upstream

### ISSUE THREAD

**Issue by evidolob** [eclipse-che/che#14648]
### Problem
Now we adopting `master` Theia branch which may contains breaking changes on each commit, build and work che-theia may be unstable. Recently we have https://github.com/eclipse/che/issues/14618 and 
Also release process may be changeling, Theia do release on last Wednesday of the month, but we may want to release che-theia in the middle of month. 

### Describe the solution you'd like
Basically I see two options:

1. Fix commit in Theia master branch each day, and have dedicated CI job to test new commits each night. But it doesn't solve release problem.

2. Use last released(tag) version for che-theia master branch. It solve release problem, but may slow down appearance new feature and fixes in che-theia from Theia.

**Comment by evidolob** [comment]
cc/ @rhopp @l0rd @benoitf

**Comment by evidolob** [comment]
cc/ @vparfonov

**Comment by benoitf** [comment]
hello,

after the call I was thinking on that and for me there is no issue.


Is che is broken if theia upstream is broken ? no. As long as che-theia images are deployed only if tests are passing, it's fine. We have only working images deployed. (che-theia:next or sidecar images shouldn't be deployed if build is failing BTW)

is che-theia may be broken ? yes but for me it's normal to break che-theia if upstream theia has changes as we need to fix in any case the issue.
We detect and fix early.

Is it a problem for a developer ? usually no as he might still work on its current version.

What happens if we're using an old version of theia ?
we will merge PR, ok but then someone will still need to fix the codebase due to upstream changes. Fixing early is the way to go for me.
And probably ppl will detect that the day of the release and it will postpone the release a lot.

Also we might add a check on theia upstream (not a blocker check but a warning check) to know even earlier that an upstream PR might break che-theia downstream.

**Comment by l0rd** [comment]
@benoitf so why https://github.com/eclipse/che/issues/14645 broke Che?

**Comment by benoitf** [comment]
@l0rd well it seems the PR check was only checking che-theia and not the endpoints so I guess some PR were merged as it was green and then images were promoted
--> https://github.com/eclipse/che/issues/14646

Also happy path should be used before publishing images as well to be sure to only publish valid content.

**Comment by l0rd** [comment]
@benoitf I agree with you. che-theia can point to theia master if PR check avoid that che-theia master (and theia) get broken. But it's not clear what do we need besides #14646: currently we do not update `next` images if the happy path PR check fails right?

**Comment by azatsarynnyy** [comment]
closing it as we decided to move with Upstream Theia PR Check https://github.com/eclipse/che/issues/15912
